### PR TITLE
Add Boilerplate Guide

### PR DESCRIPTION
**Pull request by ahijaouy**
Create a guide to walk people through setting up MERN boilerplate and connecting the repo to Heroku

### CODE REVIEW

**Review by pcshah1996 — CHANGES_REQUESTED** [review]
Line 164 has a typo, but otherwise, seems like it LGTM

**Comment by ahijaouy** [comment]
@pcshah1996 Fixed typo. 

@nareddyt if you could I think a really cool thing may be for you to pull down my branch, make a new directory called `boilerplate`, and then follow my guide and put the resulting code into that directory so it can act as like an example? Then we can move `boilerplate.md` into the `boilerplate` directory and rename it to just `README.md`? 

Kills two birds with one stone:
1. tests out the guide for flow & ease of use
2. provides a demo people can reference

**Comment by nareddyt** [comment]
### Issues

- Dead links

File: boilerplate.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+We will be using Facebook's [Create React App](cra) to help us set up the frontend of our application.
+
+1. Navigate to the root of your project repository.
+2. Run Create React App to create your frontend boilerplate
+
+    ```sh
+    npx create-react-app client
+    cd client
+    npm start
```
Comment (nareddyt) [inline boilerplate.md]: `npm start` should probably be executed after step 3 below. Doesn't make a difference in the behavior, but easier to understand.

File: boilerplate.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+## React Boilerplate
+
+We will be using Facebook's [Create React App](cra) to help us set up the frontend of our application.
+
+1. Navigate to the root of your project repository.
+2. Run Create React App to create your frontend boilerplate
+
+    ```sh
+    npx create-react-app client
+    cd client
```
Comment (nareddyt) [inline boilerplate.md]: Break this up into multiple steps. Remember some of our bootcamp members might not know what `cd client` or `npm start` actually does :)

File: boilerplate.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+## Express Boilerplate
+
+We will be using the [Express Generator](ex-gen) to quickly create our server skeleton.
+
+1. In your terminal navigate back to the root of your project respository.
+2. Use NPM to install the `express-generator` package globally on your machine
+
+   ```sh 
+   npm install express-generator -g
```
Comment (nareddyt) [inline boilerplate.md]: Global install requires `sudo` permission on UNIX. On windows, command prompt needs to be Run as Administrator.

File: boilerplate.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+   express --git --no-view server
+   cd server
+   npm install
+   npm install --dev nodemon dotenv-cli
+   ```
+
+4. We will also use `nodemon` and `dotenv-cli` to make our workflow a bit nicer.
+
+   ```sh 
+   npm install --dev nodemon dotenv-cli
```
Comment (nareddyt) [inline boilerplate.md]: Duplicate from above

File: boilerplate.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+   ```sh 
+   npm install express-generator -g 
+   ```
+
+3. Use Express Generator to create your backend boilerplate
+
+   ```sh 
+   express --git --no-view server
+   cd server
+   npm install
```
Comment (nareddyt) [inline boilerplate.md]: Split this into 2 steps

File: boilerplate.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+   npm install --dev nodemon dotenv-cli
+   ```
+
+4. We will also use `nodemon` and `dotenv-cli` to make our workflow a bit nicer.
+
+   ```sh 
+   npm install --dev nodemon dotenv-cli
+   ```
+
+5. Open the `server/package.json` where we will add a new script `dev`. This script will automatically load our environemnt variables and run our project with `nodemon` to enable hot reloading the server code. Change the `scripts` to read as follows:
```
Comment (nareddyt) [inline boilerplate.md]: Explain **hot reloading**

File: boilerplate.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
+# MERN Project Boilerplate
+
+This tutorial will take you through how to set up a new repo with the boilerplate necessary for a MERN project and how to get it deployed with Heroku.
+
+
+## Prerequisites
+
+* Before begining, make sure you've created a new repository on Github and clone it down to your local machine. It is probably a good idea to initialize this repository with the Node `.gitignore`, a `README.md`, and a `LICENSE` (i.e. GPLv3).
+* Make sure you have the latest version of [Node.js](node) installed. (Minimum: Node 8.10.0)
```
Comment (nareddyt) [inline boilerplate.md]: GitHub doesn't like these style of links, says 404 when clicking on them. @ahijaouy could you fix these?

**Review by nareddyt — COMMENTED** [review]
I tested everything up to **Putting everything together** and found these issues. I'm going to push a version of the boilerplate.md that has changes to fix these. I will also include the example files.

I did not test out deploying to heroku.

**Comment by nareddyt** [comment]
Uploaded the boilerplate example code + made a few changes to the guide. @ahijaouy please take a look when you get a chance at the changes.

**Comment by ahijaouy** [comment]
Fixed the links that were broken (oops!) and did the last step for the example code! Merging this into master :) Thanks for the review guys!!!

**Pull request merged**